generate-test-vector.py

```diff
@@ -124,50 +118,52 @@ def generate_labeled_output_tests():
     input_pub_keys = [I1, I2]
 
     recipient_bip32_seed = 'f00dbabe'
-    label_ints = [(2).to_bytes(32, 'big').hex(),(3).to_bytes(32, 'big').hex(),(1001337).to_bytes(32, 'big').hex()]
-    recipient_labels = [[(bytes.fromhex(label_int)*G).get_bytes(False).hex(), label_int] for label_int in label_ints]
     b_scan, b_spend, B_scan, B_spend = reference.derive_silent_payment_key_pair(bytes.fromhex(recipient_bip32_seed))
+    label_ints = [2, 3, 1001337]
 
     address = reference.encode_silent_payment_address(B_scan, B_spend, hrp=HRP)
     labeled_addresses = [
-        reference.create_labeled_silent_payment_address(B_scan, B_spend, bytes.fromhex(case), hrp=HRP) for case in label_ints
+        reference.create_labeled_silent_payment_address(b_scan, B_spend, case, hrp=HRP) for case in label_ints
     ]
     recipient_addresses = [address] + labeled_addresses
     comments = ["Receiving with labels: label with even parity", "Receiving with labels: label with odd parity", "Receiving with labels: large label integer"]
     for i, case in enumerate(label_ints):
         sender, recipient, test_case = new_test_case()
-        address = reference.create_labeled_silent_payment_address(B_scan, B_spend, bytes.fromhex(case), hrp=HRP)
+        address = reference.create_labeled_silent_payment_address(b_scan, B_spend, case, hrp=HRP)
         addresses = [(address, 1.0)]
 
         inputs = []
         for i, outpoint in enumerate(outpoints):
             inputs += [{
-                'prevout': list(outpoint) + [get_p2pkh_scriptsig(input_pub_keys[i], input_priv_keys[i][0]), ""],
-                'scriptPubKey': get_p2pkh_scriptPubKey(input_pub_keys[i]),
+                'txid': outpoint[0],
+                'vout': outpoint[1],
+                'scriptSig': get_p2pkh_scriptsig(input_pub_keys[i], input_priv_keys[i][0]),
+                'txinwitness': '',
+                'prevout': {'scriptPubKey': {'hex': get_p2pkh_scriptPubKey(input_pub_keys[i])}},
```

Comment:
Thank you for improving test vector structure, much better than just an array.

I have a non-blocking proposal. Should we either move `scriptSig` and `txinwitness` to `prevout`. Or alternatively pop `scriptPubKey` one layer up out of `prevout`. I'm not sure what was the idea for putting them on different levels.

Also why have only filed `hex` within `scriptPubKey` object? Seems redundant.

Feel free to merge as is and we can address afterwards

Reply:
I chose this structure to match the `vin` structure from `getrawtransaction <txid> verbosity=2`. This means we can very easily create new test cases on testnet/signet/mainnet and parse them directly into the reference script without any changes. For `scriptPubKey`, there is a hex field, whereas `txinwitness` and `scriptSig` are already only serialized as hex.

Also, conceptually, I think this is more correct: the scriptPubKey is on the `prevout` and `scriptSig` and `txinwitness` are fields on the current vin.